src/test/ui/impl-trait/static-return-lifetime-infered.stderr

```diff
@@ -0,0 +1,44 @@
+error: can't infer an appropriate lifetime
+  --> $DIR/static-return-lifetime-infered.rs:17:16
+   |
+LL |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> {
+   |                                   ----------------------- this return type evaluates to the `'static` lifetime...
+LL |         self.x.iter().map(|a| a.0)
+   |         ------ ^^^^ can't infer an appropriate lifetime
```

Comment:
Nit: this `cant infer an appropriate lifetime` hint seems out-of-place to me. it interjects in the middle of the sentence "this return type evaluates to the `'static` lifetime but this borrow...". Perhaps move it or combine it somehow with the "can't outlive the anonymous lifetime" error?

Reply:
Agree. I tried a couple of different outputs, including:

```
error: can't infer an appropriate lifetime
  --> file.rs:11:16
   |
11 |         self.x.iter().map(|a| a.0)
   |                ^^^^ can't infer an appropriate lifetime for this
   |
note: this return type evaluates to the `'static` lifetime...
  --> file.rs:10:35
   |
10 |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> {
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^
note: ...but this borrow...
  --> file.rs:11:9
   |
11 |         self.x.iter().map(|a| a.0)
   |         ^^^^^^
note: ...can't outlive the anonymous lifetime #1 defined on the method body at 10:5
  --> file.rs:10:5
   |
10 | /     fn iter_values_anon(&self) -> impl Iterator<Item=u32> {
11 | |         self.x.iter().map(|a| a.0)
12 | |     }
   | |_____^
help: you can add a constraint to make the `impl Trait` non`'static` and match the anonymous lifetime #1 defined on the method body at 10:5
   |
10 |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> + '_ {
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I feel that the output without any primary label would look reasonably good.

Reply:
I actually like what you have above, but I think you could remove the "can't infer an appropriate lifetime for this" at the top, since that's covered in the "but this borrow... can't outlive the anonymous lifetime".

Reply:
Can you take a look at the current output? I removed the primary label and it does seem slightly easier to read.